lib/std/os/linux/tls.zig

```diff
@@ -539,7 +539,7 @@ inline fn mmap(address: ?[*]u8, length: usize, prot: usize, flags: linux.MAP, fd
             prot,
             @as(u32, @bitCast(flags)),
             @as(usize, @bitCast(@as(isize, fd))),
-            @as(usize, @truncate(@as(u64, @bitCast(offset)) / linux.MMAP2_UNIT)),
+            @as(usize, @truncate(@as(u64, @bitCast(offset)) / std.heap.pageSize())),
```

Comment:
This `pageSize()` call is going to be problematic on some targets because, when this code is running, we're not necessarily allowed to make calls yet. But fortunately `offset` is always zero in this case. I suggest just simplifying this whole function down to the bare minimum actually needed for the use in this file.